ProjectsNetwork/Controllers/PostsController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using ProjectsNetwork.Data;
+using ProjectsNetwork.DataAccess.Repositories.IRepositories;
+using ProjectsNetwork.Models;
+
+// For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ProjectsNetwork.Controllers
+{
+    public class PostsController : Controller
+    {
+        private readonly IProjectRepository _projectRepository;
+
+        public PostsController(IProjectRepository projectRepository)
+        {
+            this._projectRepository = projectRepository;
+        }
+        
+        // GET: /<controller>/
+        public IActionResult Index()
+        {
+            var projects = this._projectRepository.GetAll();
+            return View(projects);
+        }
+
+        public IActionResult Post()
+        {
+            return View("PostForm");
+        }
+
+        [HttpPost]
+        public IActionResult Post(Project project)
+        {
+            project.CreationDate = DateTime.Now;
+            ClaimsPrincipal currentUser = this.User;
+            var currentUserID = currentUser.FindFirst(ClaimTypes.NameIdentifier).Value;
+            project.UserId = currentUserID;
+
+            //return Json(project);
+
+            this._projectRepository.Insert(project);
+            this._projectRepository.Save();
+            var projects = this._projectRepository.GetAll();
```

Comment:
Probably should include error checking. What if no user is found? What if the project is null? What if save throws an exception for some reason?

Reply:
got it, will look over that